python/cudf/cudf/core/reshape.py

```diff
@@ -367,3 +367,41 @@ def length_check(obj, name):
             df_list.append(col_enc_df)
 
         return concat(df_list, axis=1).drop(labels=columns)
+
+
+def merge_sorted(objs, keys=None, ascending=True, na_position="last"):
+    """Merge a list of sorted DataFrame or Series objects.
+
+    Note: Dataframes must be pre-sorted.
+
+    Parameters
+    ----------
+    objs : list of DataFrame, Series, or Index
+    keys : list, default None
+        List of Column names to sort by. If None, all columns used.
+    ascending : bool, default True
+        Sorting is in ascending order, otherwise it is descending
+    na_position : {‘first’, ‘last’}, default ‘last’
+        'first' nulls at the beginning, 'last' nulls at the end
+
+    Returns
+    -------
+    A new, lexocographically sorted, DataFrame/Series.
+    """
+
+    if not isinstance(objs, list):
+        raise TypeError("objs must be a list of Frame-like objects")
+
+    if len(objs) < 1:
+        raise TypeError("objs must be non-empty")
+
+    if len(objs) == 1:
+        return objs[0]
+
+    result = objs[0]._constructor._from_table(
+        cudf._libxx.merge.merge_sorted(
```

Comment:
Instead of directly reaching into the Cython here we should probably create a `frame` method for `merge_sorted` and then call that instead. That way `frame` is the single source of compute functions.

Reply:
No problem - I didn't do that because this is not actaully a "method,"  but I could also define it as a `classmethod` in `frame`.  Is that what you mean?

Reply:
hmm, good point. Let's just leave this as is I suppose 😄. Thanks for the explanation.